### ISSUE TITLE
Improve `context.runtime.file` when `logtail` is called indirectly

### ISSUE THREAD

**Issue by gustavopch** [logtail/logtail-js#89]
Suppose I have a file named `utils/logging.ts` containing:
```
export const log = (message, context) => {
  logtail.info(message, context)
}
```

If I import and call `log('Hello')` from a file named `foo.js`, the `context.runtime.file` is set to `utils/logging.js`. I wanted it to be `foo.js` instead.

**Comment by curusarn** [comment]
Hi @gustavopch,

Thank you for reaching out!
You can customize which file and function becomes part of the log context. 

Here's an example of file `logging.js` with simple function for logging:
```
const { Logtail } = require("@logtail/node");
const { LogLevel } = require("@logtail/types");

module.exports = {
    log(msg) {
        const token = process.env.BETTER_STACK_SOURCE_TOKEN;
        const logtail = new Logtail(token, {level: "debug"});
        const stackContextHint = { fileName: "logging", methodNames: [ "log" ] };
        logtail.log(msg, "info", { key: "value" }, stackContextHint);
    },
};
```

Note that you need to use `logtail.log()` and pass `stackContextHint` as an argument.
Could you try this out and let me know if it works for you? 🙏

**Comment by gustavopch** [comment]
Hi @curusarn, thanks for the information. I think it's missing from the docs. I tried what you suggested, but it didn't point to the line/function that I was expecting. I ended up passing the source location manually to `context`. It's super simple, looks like `new Error().stack?.split('\n')[3]?.trim()`, and works on the browser too – by inspecting this repo, it seems `@logtail/browser` doesn't include that logic.

**Comment by PetrHeinz** [comment]
Hi @gustavopch, thanks for opening the issue 🙌 

It's tru that there were some issues in how StackContextHint was handled at the moment. I've fixed them in #93 and added a usage example to our example project. I'm planning on mentioning it in the docs as well.

I hope it would work for you after the changes, works nicely in example project. 🙏 

The reason we're not using `new Error().stack` in browser environment is that the output is not standardized at the moment. Each browser has slightly different syntax. Also, code used in browser env is often going through a build process and/or minification, which often makes the results unusable.

However, if `new Error().stack` works for your use cases, feel free to continue using it in your custom context 👍

**Comment by gustavopch** [comment]
Hi @PetrHeinz, got it, thanks for the explanation.

> Also, code used in browser env is often going through a build process and/or minification, which often makes the results unusable.

FWIW, source map supported (as asked in https://github.com/logtail/logtail-js/issues/63) would help on this specific part.
